### ISSUE TITLE
403 ERROR

### ISSUE THREAD

**Issue by AliD201** [jikan-me/jikan-rest#102]
Greetings, 
Thanks for the nice API , 
while i was trying to use it for a course project today it suddenly starts to give me 
Status:403
hmmm, what's going on ?

btw i am only requesting animelists. 
Thank you.

**Comment by Zaczero** [comment]
There were some problems with Jikan being blocked by MyAnimeList.net today. In such case you have the following options:
1. wait for the block to be invalidated
2. install Jikan on your own server so you are connecting to MAL from your own/private IP address

Installing Jikan _(with docker or traditional way)_:
https://github.com/fethica/jikan-rest-docker
https://github.com/jikan-me/jikan-rest

**Comment by irfan-dahir** [comment]
Hi @AliD201 , 

We had some issues with the API earlier today. It should be working now though.
Can you confirm whether you're facing the same issue again?

**Comment by AliD201** [comment]
> Hi @AliD201 ,
> 
> We had some issues with the API earlier today. It should be working now though.
> Can you confirm whether you're facing the same issue again?

Yes now it's working! 

Thank you.

**Comment by AliD201** [comment]
@irfan-dahir 
it seems the problem is still on, as if some requests goes and some others get blocked.

**Comment by irfan-dahir** [comment]
@AliD201 I can't seem to reproduce the issue on my end. I've tested the servers' connectivity with MAL and they're not showing any issues.

Can you provide some request/response logs?

**Comment by irfan-dahir** [comment]
Closing due to inactivity. Feel free to open a separate issue if the problem occurs again.

**Comment by Jeluchu** [comment]
Hi!
First thank you for your continuous work, on the other hand comment that this error is being replicated in many calls:

**GET**: https://api.jikan.moe/v3/schedule/monday
`{
  "status": 403,
  "type": "BadResponseException",
  "message": "Something went wrong, please try again later",
  "error": "403 on https://myanimelist.net/anime/season/schedule"
}`

**GET**: https://api.jikan.moe/v3/anime/9253/videos
`  "status": 403,
  "type": "BadResponseException",
  "message": "Something went wrong, please try again later",
  "error": "403 on https://myanimelist.net/anime/9253/_/video"
}`

These two calls are some examples, but there are many more with this problem. Thank you so much for everything

Regards!

**Comment by irfan-dahir** [comment]
Hello @Jeluchu ,

MAL actually blocked every cloud provider/VPN/etc IP. This has happened before, we're communicating with MAL on the duration of this block. Last time it lasted for about 8 hours.
This is the reason any uncached requests on Jikan's servers return 403 as Jikan is unable to connect to MAL.

Additionally we're also working on a long term fix (less depending on MAL) which should be out by later this month of July.

Thanks for your patience. You can check out the #status channel on our [discord](http://discord.jikan.moe) for updates.

**Comment by Jeluchu** [comment]
Thank you very much for your quick response @irfan-dahir 

I did not want to open a new issue because I saw that it had previously happened, but I still wanted to report it so that you were aware

**Comment by irfan-dahir** [comment]
Thank you for reporting either way. Meanwhile I've re-opened and pinned this issue for anyone else that would like to follow. Will update the issue as the situation develops.

**Comment by irfan-dahir** [comment]
Hi, so here's a bit of an update.

MyAnimeList is still blocking cloud providers so Jikan is still unable to fetch the respective data. According to them, they're still looking into a solution.

This message on discord details what Jikan is doing to continue the service during this block: https://discordapp.com/channels/460491088004907029/461020417998454785/719365075172851732
tl;dr we're indexing cache into a database format to generate responses for endpoints.

The following endpoints have currently been restored
✅ Anime
✅ Manga
✅ Anime Search
✅ Manga Search
✅ Top Anime & Manga
✅ Anime & Manga Genre
✅ Producers
✅ Magazines
✅ Anime Schedule

All these endpoints are currently possible by looking up locally indexed Anime & Manga that we have salvaged based on existing local cache.

And unfortunately, we are still lacking Characters & People entries at the moment as this is all that could be salvaged from existing cache. We're looking into what can be done for the missing entries.

We're currently working on restoring the seasonal, people, characters, and all search endpoints.

### Known Issues
Note: There may be some issues since we're working on local data. Here are some known ones.

⚠Anime & Manga endpoints excludes detailed data from sub-endpoints such as characters, staff, stats, pictures etc. Which is why it's not working at the moment. 
@JanPlazovnik 

ℹ While advanced search filters have been restored although some filters may not work due to the way caching has been done up till now. These include the following filters `letter`, `genre`.

ℹ While Top endpoint filters have been restored although some filters may not work due to the way caching has been done up till now. These include the following filters `bypopularity`, `favorite`. Anime/Manga `type` filters work fine.

ℹ You may still face 403s until a missing resource has been populated from existing cache data. 
Or cache for that particular resource may not exist (we're also working on making uncached data available).

e.g https://api.jikan.moe/v3/anime/2 - This anime does not exist however since this request attempts to connect with MAL, it returns a 403 - where-as the correct response should be a 404. We're working on it.

ℹ There may be some missing entries or outdated information. We're working on updating these.


Thanks for your continued patience. I'll keep this thread updated with the information available in #status over on our [Discord channel](http://discord.jikan.moe).

**Comment by UltimateForm** [comment]
hello! season endpoint is functioning for me right now, thank you for the good work!!

**Comment by irfan-dahir** [comment]
Update:

✅ Seasonal endpoint has been completely restored (`/season`, `/season/{year}/{season}`, `/season/later`, `/season/archive`)

**Comment by irfan-dahir** [comment]
Jikan can connect to MAL again, looks like we're unblocked. 

Closing issue for now.

**Comment by barenzimo** [comment]
I can't access the API from my ip. When I'm trying to access with VPN, it's all wokring fine.
